plugin/storage/badger/factory.go

```diff
@@ -117,7 +115,9 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 		opts.ValueDir = f.Options.Primary.ValueDirectory
 
 		// These options make no sense with ephemeral data
-		opts.Truncate = f.Options.Primary.Truncate
+		if f.Options.Primary.Truncate {
```

Comment:
Since this flag has no effect anymore, we should limit its existence to `options.go`. There's no need to translate it from CLI flags to a struct, we can just warn right there.

Reply:
I have a question:
 In options.go there is no logger object. Hence to add a warning log, we need to add a logger which will need to change `Configurable` interface's `InitFromViper` signature to have a logger object and hence will change in every type that is implementing that interface. Do you think this is the right way to achieve it, I wanted to check if I am thinking in right direction.

Reply:
yes, it's fine, if you look at all InitFromViper there are some that take a logger for this reason:
```
./cmd/flags/admin.go:77:func (s *AdminServer) initFromViper(v *viper.Viper, logger *zap.Logger) {
./cmd/agent/app/reporter/flags.go:55:func (b *Options) InitFromViper(v *viper.Viper, logger *zap.Logger) *Options {
./cmd/query/app/flags.go:103:func (qOpts *QueryOptions) InitFromViper(v *viper.Viper, logger *zap.Logger) *QueryOptions {
```

Reply:
Thank you so much @yurishkuro